### DISCUSSION TITLE
Spotify playlist name as source for tags

### DISCUSSION THREAD

**Discussion by Paolami** [Marekkon5/onetagger#423] [Ideas]
Hi,

Thank you for a very interesting program! Unfortunately, it doesn't seem to be able to do what I'm looking for. I'm organizing my music into playlists in Spotify and would like to use the same sorting in my DJ program (Traktor).

If OneTagger could tag my local files with the playlist names from Spotify, that would solve my problems! Then I could use "smart lists" in Traktor based on the information in the tag with the Spotify playlist information.

Wouldn't this be a great addition to your very capable program? Is it possible to add this feature?

Greeings, Paolami

**Comment by Marekkon5** [comment]
There is no way 1T could know this information. If you have the same song in multiple playlists, it cannot know which playlist is it from. 1T also does not (and can not at the moment) read your Spotify library. 1T would also have to keep and sync your Spotify playlists locally because of rate limits and other technicalities, which is out of scope. 1T works on per-file basis, so it has no context other than the file itself. Last but not least, the feature is kind of very out of scope for 1T. So sorry, we won't be able to add this, however feel free to comment or open new issues if you have other suggestions.

**Reply by Paolami** [reply]
I understand. Thank you for your answer! I will have to try to find some other solution.

**Reply by kmlbgn** [reply]
Hi, have you found a workaround ? I am in the exact same situation.

**Reply by Paolami** [reply]
> Hi, have you found a workaround ? I am in the exact same situation.

Hi, no, unfortunately I still haven't found any solution. I still sort the songs manually (which is a pain). If you find any workaround: please tell me how!

**Comment by kmlbgn** [comment]
@Marekkon5 I was studying your comment above to see if I could implement this feature locally in a simple way. 

> There is no way 1T could know this information. If you have the same song in multiple playlists, it cannot know which playlist is it from. 1T also does not (and can not at the moment) read your Spotify library.

I'm pretty sure Spotify serves you the playlist name, rest is just a bit of code I'm guessing. What is missing is 1T feature to fetch Spotify library or parse a m3u/csv file of the playlist itself generated by 3rd party (later is simpler I guess). Or am I missing something ?

IMO, this feature could be super handy since a large part of music discovery happens on Spotify before people get the files locally. 
Like @Paolami , I also use a tag per playlist mechanism.

**Reply by Paolami** [reply]
I have actually made a python script that works and does both this and more (syncing with my Traktor (DJ) playlists) just a few days ago. Will try it out a little more and maybe develop it further a little and then publish it, but right now it:

1. Looks through my Spotify playlists and filters out the ones I want to sync
2. Looks through my local mp3 files.
3. Matches the Spotify songs and the local files.
4. Edits my Traktor playlists.

All with one click!

**Reply by kmlbgn** [reply]
Are you adding tags to each mp3 based on playlist name ?

**Reply by Paolami** [reply]
I could do that, but right now I skip that step and just add the songs directly to the Traktor playlists, which always has been my ultimate goal. :)
